### PR TITLE
Fix for deprovisioning workflow

### PR DESCRIPTION
**Pull request by sayanh**
**Description**

Changes proposed in this pull request:

- Fix for deprovisioning workflow
- Add/Fix metrics tests

### CODE REVIEW

File: components/kyma-metrics-collector/pkg/process/process.go [hunk 1]
```diff
@@ -360,8 +379,18 @@ func (p *Process) populateCacheAndQueue(runtimes *kebruntime.RuntimesPage) {
 		} else {
```
Comment (k15r) [inline components/kyma-metrics-collector/pkg/process/process.go:379]: how about removing this else block by continuing at the end of its 'if'

File: components/kyma-metrics-collector/pkg/process/process_test.go [hunk 2]
```diff
@@ -247,8 +247,8 @@ func TestPopulateCacheAndQueue(t *testing.T) {
 		g.Expect(err).Should(gomega.BeNil())
 
 		for _, failedID := range provisionedAndDeprovisionedSubAccIDs {
-			runtime := kmctesting.NewRuntimesDTO(failedID, fmt.Sprintf("shoot-%s", kmctesting.GenerateRandomAlphaString(5)), kmctesting.WithProvisionedAndDeprovisionedState)
-			runtimesPage.Data = append(runtimesPage.Data, runtime)
+			skrRuntime := kmctesting.NewRuntimesDTO(failedID, fmt.Sprintf("shoot-%s", kmctesting.GenerateRandomAlphaString(5)), kmctesting.WithProvisionedAndDeprovisionedState)
```
Comment (k15r) [inline components/kyma-metrics-collector/pkg/process/process_test.go:250]: that variable name will introduce head scratches after the next product renaming

File: components/kyma-metrics-collector/pkg/process/process.go (hunk 1 above)
Comment (sayanh) [reply components/kyma-metrics-collector/pkg/process/process.go]: removed

File: components/kyma-metrics-collector/pkg/process/process_test.go (hunk 2 above)
Comment (sayanh) [reply components/kyma-metrics-collector/pkg/process/process_test.go]: changed to `rntme`

File: components/kyma-metrics-collector/pkg/process/process.go [hunk 3]
```diff
@@ -209,8 +214,10 @@ func (p *Process) execute(identifier int) {
 
 	for {
 		var payload []byte
-		// Pick up a subAccountID to process from queue
+		// Pick up a subAccountID to process from queue and mark a Done()
```
[the remark this answers is not in the document]
Comment (sayanh) [reply components/kyma-metrics-collector/pkg/process/process.go:217]: Fixed

**Review by k15r — APPROVED** [review]

**Pull request merged**